lib/Frontend/CachedDiagnostics.cpp

```diff
@@ -82,7 +85,8 @@ struct SerializedDiagnosticInfo {
 struct SerializedFile {
   std::string FileName;
   SerializedSourceLoc IncludeLoc = SerializedSourceLoc();
-  StringRef Content;
+  bool IsFileBacked;
+  std::string Content;
```

Comment:
This is probably a silly question, but why do we need a copy here instead of a StringRef? I thought we would either have source buffer or cas buffer

Reply:
It is not a CAS buffer, but CASID, which needs a copy. Maybe I should not share those two with the same field. I will split them up.